ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/LedgerReader.scala

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.api
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.DamlLogEntryId
+import com.daml.ledger.participant.state.v1.{Configuration, LedgerId, Offset, TimeModel}
+import com.digitalasset.ledger.api.health.{HealthStatus, Healthy}
+@SuppressWarnings(Array("org.wartremover.warts.ArrayEquals"))
+case class LedgerRecord(offset: Offset, entryId: DamlLogEntryId, envelope: Array[Byte])
```

Comment:
`Array`s are only equal if they are the same reference. This might not be a problem right now, but is a potential problem. One could move the `envelope` field out of the scope of equality by putting it into a separate parameter list.

```
scala> Array(1) == Array(1)
res0: Boolean = false

scala> val arr = Array(1)
arr: Array[Int] = Array(1)

scala> arr == arr
res1: Boolean = true
```

Reply:
Alternatively, `LedgerRecord` might not need to be a case class, if it is only used as a transport object / handle for the data and doesn't need to be compared, etc...

Reply:
Switched to a plain class.